translations/i18nCheck.R

```diff
@@ -0,0 +1,66 @@
+# For developers to detect unreasonable calls in code to translate strings.
+# both applies to the R and QML code of the analysis module.
+
+library(potools)
+library(cli)
+
+checkStatus <- c()
+# Generate pot meta data from .R
+cli_h1("Check R translations:")
+rPotData <- potools::get_message_data(dir = ".",verbose = TRUE)
+
+# Get wrong usage of gettext from .R
+rErrorCalls <- subset(rPotData, rPotData$msgid=="", select = c("file", "call", "line_number"))
+          
+if (nrow(rErrorCalls) > 0) {
+  checkStatus <- c(checkStatus, 1)
+  # Warning in CLI
+  cli_alert_danger("{nrow(rErrorCalls)} empty gettext call(s) found")
+  cli_h2("Please refer to following to resolve them:")
+  print.data.frame(rErrorCalls, row.names = FALSE)
+  }else{
+    cli_alert_success("R message check PASSED")
+} 
+
+cli_h1("Check QML translations:")
+# Get QML files paths
+qmlFiles <- as.list(list.files(path = ".", pattern = ".qml", recursive = T))
+
+if (length(qmlFiles) == 0) {
+  cli_alert("No QML file found")
+  } else {
+  # Generate pot meta data from QML
+  qmlSrcData <- data.frame()
+  message("Getting QML-leve messages...")
+
+  for (i in 1:length(qmlFiles)) {
+    filePaths <- paste0("./", qmlFiles[[i]])
```

Comment:
```suggestion
    filePaths <- paste0("./", qmlFiles[i])
```
without `as.list` `[` is fine, with `as.list` `[[` is needed.